### PR TITLE
homework 05

### PR DESCRIPTION
**Pull request by darya-paliakova**

### CODE REVIEW

File: test/specs/getAdd.spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    it('should return -12 when called with numbers -12 and 0', function() {
+        expect(calculator.getAdd(-12, 0)).to.equal(-12);
+    });
+
+    it('should throw an error if provided with not numbers', function() {
+        let callWithError = () => calculator.getAdd('test', 12);
+        expect(callWithError).to.throw(`Incorrect type of data!`);
+    });
+});
```
Comment (piatrovm) [inline test/specs/getAdd.spec.js:22]: no new line added

File: test/specs/getMultiply.spec.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    it('should return 0 when called with numbers 6 and 0', function() {
+        expect(calculator.getMultiply(6, 0)).to.equal(0);
+    });
+
+    it('should throw an error if provided with not numbers', function() {
+        let callWithError = () => calculator.getMultiply('test', 12);
+        expect(callWithError).to.throw(`Incorrect type of data!`);
+    });
+});
```
Comment (piatrovm) [inline test/specs/getMultiply.spec.js:30]: no new line added

File: app/calculator.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    }
+
+    getMultiply(firstNumber, secondNumber){
+        if((typeof firstNumber) !== 'number' || (typeof secondNumber) !== 'number'){
+            throw new Error(`[${firstNumber}] OR [${secondNumber}] are not numbers.`);
+        } else {
+            return firstNumber * secondNumber;
+        }
+    }
+}
```
Comment (piatrovm) [inline app/calculator.js]: you need to use `module.exports = Calculator;` to use this class in *.spec.js

File: test/specs/getAdd.spec.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const Calculator= require('../../app/calculator.js');
+const chai = require('chai');
+const spies = require('chai-spies');
+const { expect } = require('chai').expect;
```
Comment (piatrovm) [inline test/specs/getAdd.spec.js]: incorrect import. True option: `const { expect } = require('chai');`

File: test/specs/getAdd.spec.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+chai.use(spies);
```
Comment (piatrovm) [inline test/specs/getAdd.spec.js]: You've copypasted from tutorial but you never use this in your code

File: test/specs/getAdd.spec.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+describe('getAdd', () => {
+
+
+    it('should return 23 when called with numbers 12 and 11', () => {
+        expect(calculator.getAdd(12, 11)).to.equal(23);
```
Comment (piatrovm) [inline test/specs/getAdd.spec.js:11]: incorrect chai syntax. Pleas read documentation. Correct `to.be.equal()`

File: test/specs/getAdd.spec.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+const Calculator= require('../../app/calculator.js');
+const chai = require('chai');
+const spies = require('chai-spies');
+const { expect } = require('chai').expect;
+chai.use(spies);
+
+describe('getAdd', () => {
+
```
Comment (piatrovm) [inline test/specs/getAdd.spec.js:8]: no class expression `let calculator = new Calculator();`

**Comment by piatrovm** [comment]
Hi Darya, please review all my comments:
- if you use a project draft then please delete code which is not for this task (I mean Factorial class and its tests)
- make project private
- read more about chai syntax and use it appropriately
- remember class declaration, exporting it and using in other code (import-export)
- also re-read information about class expression and so on

And remember, if you copy something it is ok, but please do it more accurate

File: app/calculator.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+class Calculator{
+    constructor() {
```
Comment (piatrovm) [inline app/calculator.js]: no need in empty constructor(){}

**Comment by piatrovm** [comment]
Your tests have no `const calculator = new Calculator();`. Do they work?

**Comment by darya-paliakova** [comment]
@piatrovm could you please review the pull request?

**Comment by piatrovm** [comment]
@darya-paliakova now looks good, ok to merge

**Pull request merged**